demo.py

```diff
@@ -13,212 +13,96 @@
+    parser.add_argument('--verbose', action='store_true')
 
-    print_accuracy(y_train, y_train4, y_test, y_test4)
+    subparsers = parser.add_subparsers(title='dataset', description='dataset to use', dest='dataset', required=True)
 
+    sp_blobs = subparsers.add_parser('blobs', help='blobs data set')
+    sp_blobs.add_argument('--num-samples', type=int, default=2000, help='number of samples (default: %(default)s)')
+    sp_blobs.add_argument('--num-features', type=int, default=10, help='number of features (default: %(default)s)')
+    sp_blobs.add_argument('--num-informative', type=int, default=2, help='number of informative features (default: %(default)s)')
 
-    print()
-    print('=' * 28)
-    print("{:14}{:7}{:7}".format("Method", "Train", "Test"))
-    print('-' * 28)
-    for name, _y_train_pred, _y_test_pred in zip(["Adaboost", "DecisionTree", "QBoost", "QBoostIt"],
-                                                 [y_train_pred, y_train_pred2, y_train_dw, y_train4],
-                                                 [y_test_pred, y_test_pred2, y_test_dw, y_test4]):
-        print("{:14}{:<7.2f}{:<7.2f}".format(name, metric(y_train, _y_train_pred), metric(y_test, _y_test_pred)))
-    print('=' * 28)
+    sp_digits = subparsers.add_parser('digits', help='handwritten digits data set')
+    sp_digits.add_argument('--digit1', type=int, default=0, choices=range(10), help='first digit to include (default: %(default)s)')
+    sp_digits.add_argument('--digit2', type=int, default=1, choices=range(10), help='second digit to include (default: %(default)s)')
+    sp_digits.add_argument('--plot-digits', action='store_true', help='plot a random sample of each digit')
 
+    args = parser.parse_args()
 
-if __name__ == '__main__':
-    import argparse
 
-    parser = argparse.ArgumentParser()
-    parser.add_argument("dataset", choices=["mnist", "wisc"], help="data set to analyze")
-    parser.add_argument("--verbose", action="store_true", help="display additional output")
+    if args.dataset == 'blobs':
+        n_samples = args.num_samples
+        n_features = args.num_features
+        n_informative = args.num_informative
 
-    args = parser.parse_args()
+        X, y = make_blob_data(n_samples=n_samples, n_features=n_features, n_informative=n_informative)
 
-    if args.dataset == 'mnist':
+        X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.4)
 
-        print('MNIST handwritten digits data set:')
+        # See Boyda et al. (2017), Eq. (17) regarding normalization
+        normalized_lambdas = np.linspace(0.0, 0.5, 10)
+        lambdas = normalized_lambdas / n_features
+        qboost, lam = qboost_lambda_sweep(X_train, y_train, lambdas, verbose=args.verbose)
 
-        mnist = fetch_openml('mnist_784', version=1)
+        print('best lambda:', lam)
```

Comment:
print('Best lambda:'
print('Informative features:
print('Selected features:                 ---> I also suggest removing the preceding spaces, I don't find that more readable in the output